src/runtime_src/core/pcie/linux/scan.h

```diff
@@ -61,20 +61,22 @@ class pci_device {
         std::string& err_msg, std::vector<uint64_t>& iv);
     void sysfs_get(const std::string& subdev, const std::string& entry,
         std::string& err_msg, std::string& s);
-    void sysfs_get(const std::string& subdev, const std::string& entry,
-        std::string& err_msg, bool& b);
     void sysfs_get(const std::string& subdev, const std::string& entry,
         std::string& err_msg, std::vector<char>& buf);
     template <typename T>
     void sysfs_get(const std::string& subdev, const std::string& entry,
-        std::string& err_msg, T& i) {
+        std::string& err_msg, T& i, const int& default_val) {
         std::vector<uint64_t> iv;
 
         sysfs_get(subdev, entry, err_msg, iv);
         if (!iv.empty())
             i = static_cast<T>(iv[0]);
         else
-            i = static_cast<T>(-1); // default value
+            i = static_cast<T>(default_val); // default value
+    }
+    void sysfs_get_sensor(const std::string& subdev, const std::string& entry,
```

Comment:
it seems that caller don't need to pass in err_msg, don't care about error for sensor readings. if we can't get it, we print out N/A
BTW, is -1 a good default value for sensor readings? should we use 0? we should print out N/A when we get default value.

Reply:
The default value is what the user is passing. We aren't specifying 0 or -1 here. 
Should I remove the err_msg param? Wouldn't it be better if we keep it so that the caller doesn't get confused when using the template or one of the other functions? Instead, I can return "N/A" in the err_msg